src/lang/processor.rs

```diff
@@ -0,0 +1,1181 @@
+                        ProcessorError::InvalidLiteralFormat {
+                            kind,
+                            value,
+                            span: Some(get_span(&inner_lit)),
+                        }
+                    }
+                    ProcessorError::Internal(message) => ProcessorError::InvalidLiteralFormat {
+                        kind: format!("raw hex processing (internal: {})", message),
+                        value: full_literal_str.to_string(),
+                        span: Some(get_span(&inner_lit)),
+                    },
+                    _ => ProcessorError::InvalidLiteralFormat {
+                        kind: "raw hex processing error".to_string(),
+                        value: full_literal_str.to_string(),
+                        span: Some(get_span(&inner_lit)),
+                    },
+                })
+                .map(Value::from)
+        }
+        Rule::literal_string => Ok(Value::from(parse_pest_string_literal(&inner_lit)?)),
+        Rule::literal_array => {
+            let elements: Result<Vec<Value>, ProcessorError> = inner_lit
+                .into_inner()
+                .map(|elem_pair| process_literal_value(&elem_pair, context_stmt_name))
+                .collect();
+            let middleware_array = middleware::containers::Array::new(elements?)
+                .map_err(|e| ProcessorError::Internal(format!("Failed to create Array: {}", e)))?;
+            Ok(Value::from(middleware_array))
+        }
+        Rule::literal_set => {
+            let elements: Result<HashSet<Value>, ProcessorError> = inner_lit
+                .into_inner()
+                .map(|elem_pair| process_literal_value(&elem_pair, context_stmt_name))
+                .collect();
+            let middleware_set = middleware::containers::Set::new(elements?)
+                .map_err(|e| ProcessorError::Internal(format!("Failed to create Set: {}", e)))?;
+            Ok(Value::from(middleware_set))
+        }
+        Rule::literal_dict => {
+            let pairs: Result<HashMap<Key, Value>, ProcessorError> = inner_lit
+                .into_inner()
+                .map(|dict_entry_pair| {
+                    let mut entry_inner = dict_entry_pair.clone().into_inner();
+                    let key_pair = entry_inner.next().unwrap();
+                    let val_pair = entry_inner.next().unwrap();
+                    let key_str = parse_pest_string_literal(&key_pair)?;
+                    let val = process_literal_value(&val_pair, context_stmt_name)?;
+                    Ok((Key::new(key_str), val))
+                })
+                .collect();
+            let middleware_dict = middleware::containers::Dictionary::new(pairs?).map_err(|e| {
+                ProcessorError::Internal(format!("Failed to create Dictionary: {}", e))
+            })?;
+            Ok(Value::from(middleware_dict))
+        }
+        _ => Err(ProcessorError::RuleMismatch {
```

Comment:
We call `process_literal_value` after checking that the pair rule is `literal_value` which doesn't have more cases by definition, so this should be `unreachable`.